### DISCUSSION TITLE
Help me out here

### DISCUSSION THREAD

**Discussion by OM3D** [airsdk/Adobe-Runtime-Support#2759] [General]
Hello all,

I do not consider myself as an AS developer, nor I was! However, I had the pleasure of using AS in my personal little projects and I was enjoying being part of Wonderfl.net (I know it is long gone! and the reason I am making this topic is that I missed it!) 

My question is that, is there anything like it? Is AS dead and gone! Is there any other community like "Wonderfl.net"?

I used to use Flashdevelop but it seems they are also stopped developing! 
I used to know a little about AWAY3D as well. but it looks they are gone too! 

In what ways I can enjoy being a part of community -like Wonderfl- and continue doing some little projects of my own? Should I switch to Javascript? 

I also like 3D stuff so, is it good to come back to AWAY3D or should I use three.js? 

I have done a search and only found frozen -dead- flash communities.

I appreciate your suggestions in advance! :-)

- OM3D

**Comment by Ender22** [comment]
Thanks to Harman owning Adobe Air now (and doing a great job of updating it frequently and listening to the community), and an awesome foundation of community tools like Starling, and Feathers, there's still a bunch of us that choose Air and AS as our tools. My company for example has built a b2b SaaS 'behavioral tech' platform on top of Air with features such as training modules, forums, game elements, news feed, surveys, etc

Also there is great ANE support thanks to distriqt. I love them!

Don't search for 'flash' communities, but search for 'adobe air' instead since flash itself is dead. 

Here's some links!
https://gamua.com/starling/
https://feathersui.com/samples/as3-starling/
https://airnativeextensions.com/

Sorry not sure what to recommend for 3d stuff tho

**Comment by M4rkY4** [comment]
Hi,


first of all, Flash (with the help of AIR) is an amazing technology, it's a joy to develop with, it's mature, stable, very user-friendly and it's still evolving. The combination of Adobe Animate and Harman AIR is unprecedentedly fast and convenient, you just develop 2d games twice or three times faster with it! There's literally nothing like it. Starling is another great and mature Flash-based technology that allows very performant graphics in Flash / AIR.

So if you have nice experience with either of Adobe Animate / Starling or you plan to use it for few hours a month, you can still stick with Flash, it's great. Just use Intellij IDEA instead of FlashDevelop and you will do great. This Github community and Starling community at https://forum.starling-framework.org/ are active enough for fixing technical issues, but less as an art sharing/encouragement community.


But if you have little experience with Flash / Adobe Animate / Starling and you are are planning to develop few hours a day, I would advice you to think about selecting technology that will give you that kind of useful experience, you can use to switch to a dev or a freelance job in case something happens with your current one, it's a smart thing to consider.

While there are great technologies like Godot and Defold (nice for a small hobby), I believe, the safest way is to learn something the industry is massively using, like Unity (games industry) or something with JavaScript / TypeScript (web dev industry).

Unity is nice, but it's a bit too overcomplicated in almost every aspect, including the C# itself. While I like it, I would not advice it for a non-experienced person, if programming is not your main field of professional interest. It's way less intuitive than Flash.

As of a TypeScript, it's very similar to ActionScript, switching will be relatively easy. PixiJS looks very promising ( https://pixijs.com/ ), especially with the future WebGPU support. Runtime speed of JS/TS is way better than AS3. It looks easy to learn, but large caveats are that sooner or later you will have to learn how console NPM and relative non-UI things works and you will have trouble with mobile games export.

Just don't expect as user-friendly ecosystem as Flash has from Unity or Pixi. Godot/Defold seem to be better in this aspect.


To sum up my advice:
experience with Flash / Adobe Animate / Starling and only 2D - use Flash and Intellij IDEA
small hobby - use Godot, its community is quite proactive https://godotengine.org/community/
big hobby - use Unity for 3D or PixiJS for 2D
